Readme.md

```diff
@@ -4,20 +4,30 @@
 [![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
 <!-- default badges end -->
 <!-- default file list -->
-*Files to look at*:
 
-* [Default.aspx](./CS/WebSite/Default.aspx) (VB: [Default.aspx](./VB/WebSite/Default.aspx))
-* [Default.aspx.cs](./CS/WebSite/Default.aspx.cs) (VB: [Default.aspx.vb](./VB/WebSite/Default.aspx.vb))
-<!-- default file list end -->
-# Bind a grid to a DataTable via code
+# ASP.NET - Bind ASPxGridView to a DataTable object
+
+This example handles the [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event to bind [`ASPxGridView`](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView) to a programmatically created [`DataTable`](https://docs.microsoft.com/en-us/dotnet/api/system.data.datatable) object.
+
 <!-- run online -->
 **[[Run Online]](https://codecentral.devexpress.com/e168/)**
 <!-- run online end -->
 
+## Files to Look At
+
+* [Default.aspx](./CS/WebSite/Default.aspx) (VB: [Default.aspx](./VB/WebSite/Default.aspx))
+* [Default.aspx.cs](./CS/WebSite/Default.aspx.cs) (VB: [Default.aspx.vb](./VB/WebSite/Default.aspx.vb))
+
+## Implementation Details
+
+In this example, [`ASPxGridView`](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxGridView) uses a programmatically created [`DataTable`](https://docs.microsoft.com/en-us/dotnet/api/system.data.datatable) object as its data source. To bind the grid to the [`DataTable`](https://docs.microsoft.com/en-us/dotnet/api/system.data.datatable) object, the [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event handler assigns this table to the grid's [`DataSource`](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxDataWebControlBase.DataSource) property. 
+
+The example stores the initially created [`DataTable`](https://docs.microsoft.com/en-us/dotnet/api/system.data.datatable) object in [session state](https://docs.microsoft.com/en-us/previous-versions/aspnet/ms178581(v=vs.100)). This technique allows you not to recreate the table for subsequent [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event triggers.
```

Comment:
~"#Wordiness_SentenceStructure" I'd remove `initially` as it doesn't add anything to the sentence - we're just referring to the same object we created.

~"#TermUsage_TechnicalSide"? As for `triggers`, are you referring to the things that trigger/raise the event (ASPxGridView1.DataBind();) or event occurrences (so we use session state and don't recreate the object on each event occurrence)? Please clarify. I based my suggestion on the latter.

> for subsequent [`DataBinding`](https://docs.microsoft.com/en-us/dotnet/api/system.web.ui.control.databinding) event **triggers**

Reply:
I've used `initially` to focus a user's attention on the fact that the table is created only once and isn't recreated later. So your suggestion `To persist the object between...` is technically correct, but I've tried to emphasize the lack of the recreate action for better performance. In my opinion, this is the main idea of keeping the table in session state in this example.
```
DataTable GetTable() {
  //You can store a DataTable in the session state
  DataTable table = Session["Table"] as DataTable;
  if (table == null) {
      table = new DataTable();
      table.Columns.Add("id", typeof(int));
      table.Columns.Add("data", typeof(String));
      for (int n = 0; n < 100; n++) {
          table.Rows.Add(n, "row" + n.ToString());
      }
      Session["Table"] = table;
  }
```
What do you think about it?


Regarding `triggers`, your second option is what I meant - occurrences. Is `triggers` an inappropriate term?

Reply:
I just suggest to omit `initially` and see if anything changes when you read the sentence. Do we need this adverb as a modifier to `created`? I think that without the modifier the sentence reads clearer, as if we're referring to the same (created) object.

> In my opinion, this is the main idea of keeping the table in session state in this example.

I see your point regarding the main idea. How about this addition? `This technique allows you not to recreate the object each time the DataBinding event fires.`

> Regarding `triggers`, your second option is what I meant - occurrences. Is `triggers` an inappropriate term?

Stick with the `occurrence` term then - as I described above, an event trigger is what triggers/raises an event, not its occurrence.

Reply:
> I think that without the modifier the sentence reads clearer, as if we're referring to the same (created) object.

Okay, along with the mention that we are "preserving this object between `DataBinding` event occurrences", I agree that this is clear enough

---

> I see your point regarding the main idea. How about this **addition**? `This technique allows you not to recreate the object each time the DataBinding event fires.`

Do you suggest adding the above sentence after the following part?   
> To persist the object between `DataBinding` event occurrences, the example stores it in **session state**.

In this example, the grid's `DataBinding` event handler assigns a programmatically created `DataTable` object to the grid's `DataSource` property. To persist the object between `DataBinding` event occurrences, the example stores it in **session state**. This technique allows you not to recreate the object each time the `DataBinding` event fires.

---

> Stick with the occurrence term then - as I described above, an event trigger is what triggers/raises an event, not its occurrence.

I see, thank you!

Reply:
> Do you suggest adding the above sentence after the following part?

Exactly.

Reply:
Okay, thanks! I'll make the changes.